plugins/org.yakindu.sct.simulation.ui/src/org/yakindu/sct/simulation/ui/view/ExecutionContextViewerFactory.java

```diff
@@ -86,6 +89,60 @@ public void beforeEditorActivated(ColumnViewerEditorActivationEvent event) {
 					}
 				});
 
+		parent.addControlListener(new TreeViewerColumnResizer(viewer, nameColumn, valueColumn));
+
 		return viewer;
 	}
+	
+	
+	/**
+	 * 
+	 * @author robert rudi - Initial contribution and API
+	 * 
+	 */
+	private static final class TreeViewerColumnResizer extends ControlAdapter {
+
+		public static final int VALUE_COL_VIEW_WIDTH_PERCENTAGE = 3; //33%
+		public static final int VALUE_COL_MIN_WIDTH = 80;
+		public static final int NAME_COL_MIN_WIDTH = 100;
+		
+		private final TreeViewer viewer;
+		private final TreeViewerColumn nameColumn;
+		private final TreeViewerColumn valueColumn;
+		
+		private TreeViewerColumnResizer(TreeViewer viewer, TreeViewerColumn nameColumn, TreeViewerColumn valueColumn) {
+			this.nameColumn = nameColumn;
+			this.viewer = viewer;
+			this.valueColumn = valueColumn;
+		}
+		@Override
+		public void controlResized(ControlEvent e) {
+			resizeViewerColumns(viewer);
+		}
+		protected void resizeViewerColumns(final TreeViewer viewer) {
+			Tree tree = viewer.getTree();
+			Rectangle area = tree.getParent().getParent().getClientArea();
+			Point size = tree.computeSize(SWT.DEFAULT, SWT.DEFAULT);
+			ScrollBar vBar = tree.getVerticalBar();
+			int width = area.width - tree.computeTrim(0, 0, 0, 0).width + vBar.getSize().x;
+			if (size.y > area.height + tree.getHeaderHeight()) {
+				width -= vBar.getSize().x;
+			}
+			Point oldSize = tree.getSize();
+			if (oldSize.x > area.width) {
+				setColumnWidths(vBar, width);
+				tree.setSize(area.width, area.height);
+			} else {
+				tree.setSize(area.width, area.height);
+				setColumnWidths(vBar, width);
+			}
+		}
+		protected void setColumnWidths(ScrollBar vBar, int width) {
```

Comment:
I like this, except the point that the value and name columns will always be scaled to ⅓ / ⅔ when the size of the whole value view thing is changed, no matter how the user set it up before. I think it is a good initial value but overriding the user's choices on each resizing is kind of drastic - especially because this also appplies when the **window** is resized. Maybe it would be better to keep the user's proportions. You never know what kind of long enum literal someone wants to display in the value column.

Reply:
I've built in your suggestion in the following commit. The user's proportions should now be taken into account, even if the window itself is resized.